packages/trader/src/Stores/Modules/Trading/Actions/start-date.ts

```diff
@@ -1,32 +1,25 @@
 import { ContractType } from 'Stores/Modules/Trading/Helpers/contract-type';
 import { TTradeStore } from 'Types';
 
-type TOnChangeStartDate = (
-    store: TTradeStore
-) => Promise<
-    Pick<
-        TTradeStore,
-        | 'contract_start_type'
-        | 'duration_units_list'
-        | 'duration_min_max'
-        | 'duration_unit'
-        | 'start_time'
-        | 'expiry_date'
-        | 'expiry_type'
-    > & { sessions?: TTradeStore['sessions'] }
+type TOnChangeStartDate = (store: TTradeStore) => Promise<
+    Pick<TTradeStore, 'contract_start_type' | 'duration_units_list' | 'start_time' | 'expiry_date' | 'expiry_type'> & {
+        sessions?: TTradeStore['sessions'];
+    } & {
+        duration_min_max: { [key: string]: { min: number; max: number } } | { min: number; max: number };
+    } & { duration_unit: string | number }
 >;
 type TOnChangeExpiry = (store: TTradeStore) => Promise<{
-    expiry_time: TTradeStore['expiry_time'];
-    market_open_times?: TTradeStore['market_open_times'];
-    market_close_times?: TTradeStore['market_close_times'];
+    expiry_time: TTradeStore['expiry_time'] | moment.Moment;
+    market_open_times?: TTradeStore['market_open_times'] | { open: string[]; close: string[] };
```

Comment:
@kate-deriv please update these types in TOverrideTradeStore instead

Reply:
I think I'll remove that type because TS is handling returned type correctly even without type:
```
const onChangeExpiry: (store: TTradeStore) => Promise<{
    market_close_times: string[] | TTimes;
    market_open_times: string[] | TTimes;
    expiry_time: string | moment.Moment | null;
}>
```